app/gui2/src/components/GraphEditor/upload.ts

```diff
@@ -149,38 +149,43 @@ export class Uploader {
 
   private async ensureDataDirExists() {
     const exists = await this.dataDirExists()
-    if (!exists) {
-      await this.rpc.createFile({
-        type: 'Directory',
-        name: DATA_DIR_NAME,
-        path: { rootId: this.projectRootId, segments: [] },
-      })
-    }
+    if (exists) return
+    await this.rpc.createFile({
+      type: 'Directory',
+      name: DATA_DIR_NAME,
+      path: { rootId: this.projectRootId, segments: [] },
+    })
   }
 
   private async dataDirExists(): Promise<boolean> {
     try {
       const info = await this.rpc.fileInfo(this.dataDirPath())
       return info.attributes.kind.type == 'Directory'
-    } catch (err: any) {
-      if (err.cause && err.cause instanceof RemoteRpcError) {
-        if ([ErrorCode.FILE_NOT_FOUND, ErrorCode.CONTENT_ROOT_NOT_FOUND].includes(err.cause.code)) {
+    } catch (error) {
+      if (
+        typeof error === 'object' &&
+        error &&
+        'cause' in error &&
+        error.cause instanceof RemoteRpcError
+      ) {
+        if (
+          error.cause.code === ErrorCode.FILE_NOT_FOUND ||
+          error.cause.code === ErrorCode.CONTENT_ROOT_NOT_FOUND
+        )
           return false
-        }
       }
-      throw err
+      throw error
     }
   }
 
   private async pickUniqueName(suggestedName: string): Promise<string> {
     const files = await this.rpc.listFiles(this.dataDirPath())
     const existingNames = new Set(files.paths.map((path) => path.name))
-    const [stem, maybeExtension] = splitFilename(suggestedName)
-    const extension = maybeExtension ?? ''
+    const { name, extension = '' } = splitFilename(suggestedName)
```

Comment:
`stem` was better, because it was more precise. See https://doc.rust-lang.org/std/path/struct.Path.html#method.file_stem or https://en.cppreference.com/w/cpp/filesystem/path/stem